src/gc/gc.d

```diff
@@ -2759,19 +2763,21 @@ struct Gcx
                 Lnotfree:
                     p = pool.baseAddr + pn * PAGESIZE;
                     for (u = 0; u < PAGESIZE; u += size)
-                    {   biti = bitbase + u / 16;
-                        if (pool.freebits.test(biti))
-                        {
-                            List *list = cast(List *)(p + u);
-                            if (list.next != bucket[bin])       // avoid unnecessary writes
-                                list.next = bucket[bin];
-                            list.pool = pool;
-                            bucket[bin] = list;
-                        }
+                    {
+                        biti = bitbase + u / 16;
+                        if (!pool.freebits.test(biti))
+                            continue;
+                        auto elem = cast(List *)(p + u);
+                        elem.pool = pool;
+                        *tail[bin] = elem;
+                        tail[bin] = &elem.next;
                     }
```

Comment:
I see a much faster marking for the testgc3 benchmark and I think that's because the prefetcher can handle the access pattern better.
But the more important part of this pull is that it helps to reduce fragmentation by first allocating objects from first pool. Maybe we should make sure to iterate from the oldest to the newest pool here.